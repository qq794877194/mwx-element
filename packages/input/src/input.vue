<template>
<div>
    <input :type="type" v-bind="$attrs" @input="handleInput">
</div>
</template>

<script>
    export default {
        name: "mwxInput",
        inheritAttrs: false,
        props: {
            type: {
                type: String,
                default: 'text'
            },
            value: {
                type: String
            }
        },
        methods: {
            handleInput(e){
                this.$emit('input',e.target.value);
                const findParent = parent => {
                    while(parent){
                        if(parent.$options.name === 'mwxFormItem'){
                            break;
                        }
                        parent = parent.$parent;
                    }
                    return parent;
                };
                const parent = findParent(this.$parent);
                if(parent){
                    parent.$emit('validate');
                }
            }
        }
    }
</script>

<style scoped>

</style>
