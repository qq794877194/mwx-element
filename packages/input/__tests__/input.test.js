import input from '../src/input.vue';
import { mount } from '@vue/test-utils';

describe('max-input1', ()=> {
   test('input-text', () => {
       const wrapper = mount(input);
       expect(wrapper.html()).toContain('input type="text"');
   });

    test('input-password', () => {
        const wrapper = mount(input,{propsData: {type: 'password'}});
        expect(wrapper.html()).toContain('input type="password"');
    });

    test('input-value', () => {
        const wrapper = mount(input,{propsData: {type: 'password', value: 'admin'}});
        expect(wrapper.props('value')).toBe('admin');
    });

    test('input-snapshot', () => {
        const wrapper = mount(input,{propsData: {type: 'password', value: 'admin'}});
        expect(wrapper.vm.$el).toMatchSnapshot();
    });
});
