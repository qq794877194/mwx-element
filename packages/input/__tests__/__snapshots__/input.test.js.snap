// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`max-input1 input-snapshot 1`] = `
<div>
  <input
    type="password"
  />
</div>
`;
